diff --git a/uboot/u-boot-2020.04/include/configs/ax650_emmc.h b/uboot/u-boot-2020.04/include/configs/ax650_emmc.h
index 3413084e..d9c2cb5f 100755
--- a/include/configs/ax650_emmc.h
+++ b/include/configs/ax650_emmc.h
@@ -51,14 +51,16 @@
 #undef BOOTARGS_EMMC
 #endif
 
+#ifndef ROOTFS_PARTITION_SIZE
+#define ROOTFS_PARTITION_SIZE   "30380032K"
+#endif
+
 #define RECOVERY_ROOTFS_PART   "/dev/mmcblk0p12"
 
 #define EMMCPARTS_RECOVERY	EMMCPARTS_DEFAULT_POST \
     "1M(recovery_dtb)," \
     "74M(recovery)," \
-    "4736M(rootfs)," \
-    "384M(soc)," \
-    "1900M(opt)," \
+    ROOTFS_PARTITION_SIZE"(rootfs)"
 
 #define BOOTARGS_EMMC BOOTARGS_EMMC_NO_PART \
 				RECOVERY_ROOTFS_PART \
@@ -69,6 +71,12 @@
 
 #define RECOVERY_BOOTARGS BOOTARGS_EMMC_RECOVERY_NO_PART \
 				EMMCPARTS_RECOVERY
+
+
+#define CONFIG_EXTRA_ENV_SETTINGS		\
+	"seti2c0m=mw 0x4250688 0x4001\0" \
+	"before_boot_cmd=run seti2c0m\0" \
+
 #endif
 
 #endif
diff --git a/uboot/u-boot-2020.04/include/configs/ax650_common.h b/uboot/u-boot-2020.04/include/configs/ax650_common.h
index 6dc69787..efa9a448 100755
--- a/include/configs/ax650_common.h
+++ b/include/configs/ax650_common.h
@@ -368,5 +368,7 @@ mtdparts=spi5.0:4M(uboot),768K(env),1M(atf),1M(dtb),32M(kernel),512K(param),192M
 	"     setenv bootpart A; " \
 	"  fi; setenv bootcount 0; saveenv; " \
 	"  run bootcmd\0" \
+	"seti2c0m=mw 0x4250688 0x4001\0" \
+	"before_boot_cmd=run seti2c0m\0" \
 
 #endif
--- a/cmd/axera/boot/axera_boot.c	2025-12-05 12:01:11.473407109 +0800
+++ b/cmd/axera/boot/axera_boot.c	2025-12-05 11:59:20.795244876 +0800
@@ -327,6 +327,10 @@
 	const char *x_kernel = "kernel";
 	const char *x_dtb = "dtb";
 
+	char *axera_boot_after_cmd = env_get("before_boot_cmd");
+	if(NULL != axera_boot_after_cmd) 
+		run_command(axera_boot_after_cmd, 0);
+
 #ifdef CONFIG_AXERA_SUPPORT_AB_OTA
 	char *bootpart = env_get("bootpart");
 
--- a/include/configs/ax650_common.h	2025-12-05 17:56:40.850901694 +0800
+++ b/include/configs/ax650_common.h	2025-12-05 17:56:35.466796142 +0800
@@ -242,9 +242,9 @@
 #endif
 
 #define EMMCPARTS_DEFAULT	EMMCPARTS_DEFAULT_POST \
-    "4736M(rootfs)," \
-    "384M(soc)," \
-    "1900M(opt)," \
+	"1M(recovery_dtb)," \
+	"74M(recovery)," \
+    "29668M(rootfs)" 
 
 #define EMMCPARTS_MASTER	EMMCPARTS_DEFAULT_POST \
 	"2048M(rootfs)," \
